### PR TITLE
Added booking tabs, type fixing and refactoring

### PR DESCRIPTION
**Pull request by zomars**
- Adds WIP for booking tabs
- Refactored NavTabs
- Type fixes
- General refactoring

### CODE REVIEW

**Review by pumfleet — APPROVED** [review]
Great work @zomars, this is a huge refactor

**Comment by PeerRich** [comment]
EPIC work, let's sort this tab by what's coming next vs what's latest (reverse the order)

image

**Review by PeerRich — APPROVED** [review]
epic stuff @KATT gonna love the refactor, too

File: pages/bookings/[status].tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        )}
+      </td>
+      <td className="hidden sm:table-cell px-6 py-4 whitespace-nowrap">
+        <div className="text-sm text-gray-900">{dayjs(booking.startTime).format("D MMMM YYYY")}</div>
+        <div className="text-sm text-gray-500">
+          {dayjs(booking.startTime).format("HH:mm")} - {dayjs(booking.endTime).format("HH:mm")}
+        </div>
+      </td>
+      <td className="px-6 py-4 whitespace-nowrap text-right text-sm font-medium">
+        {isUpcoming && !isCancelled ? (
```
Comment (PeerRich) [inline pages/bookings/[status].tsx:114]: this could be `isUpcoming && !isCancelled && {stuff}` instead of `? <stuff> : null` right?

**Review by PeerRich — APPROVED** [review]

File: pages/bookings/[status].tsx (hunk 1 above)
Comment (zomars) [reply pages/bookings/[status].tsx:114]: Yeah. That can work too

**Pull request merged**